main.go

```diff
@@ -1,98 +1,50 @@
 package main
 
 import (
-	"errors"
 	"fmt"
 	"os"
 	"strings"
 
 	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/command/git"
 	"github.com/bitrise-io/go-utils/log"
-	"github.com/bitrise-io/go-utils/retry"
-	"github.com/bitrise-io/steps-git-clone/gitutil"
 )
 
 const (
 	retryCount = 2
-	waitTime   = 5 //seconds
+	waitTime   = 5 // seconds
 )
 
-// ConfigsModel ...
-type ConfigsModel struct {
-	CloneIntoDir  string
-	RepositoryURL string
-	Commit        string
-	Tag           string
-	Branch        string
-	CloneDepth    string
-
-	PullRequestURI         string
-	PullRequestID          string
-	BranchDest             string
-	PullRequestMergeBranch string
-	ResetRepository        string
-
-	BuildURL         string
-	BuildAPIToken    string
-	UpdateSubmodules string
-	ManualMerge      string
-}
-
-func createConfigsModelFromEnvs() ConfigsModel {
-	return ConfigsModel{
-		CloneIntoDir:  os.Getenv("clone_into_dir"),
-		RepositoryURL: os.Getenv("repository_url"),
-		Commit:        os.Getenv("commit"),
-		Tag:           os.Getenv("tag"),
-		Branch:        os.Getenv("branch"),
-		CloneDepth:    os.Getenv("clone_depth"),
-
-		PullRequestURI:         os.Getenv("pull_request_repository_url"),
-		PullRequestID:          os.Getenv("pull_request_id"),
-		BranchDest:             os.Getenv("branch_dest"),
-		PullRequestMergeBranch: os.Getenv("pull_request_merge_branch"),
-		ResetRepository:        os.Getenv("reset_repository"),
-		ManualMerge:            os.Getenv("manual_merge"),
+var (
+	config Config
+	// Git ...
+	Git         *git.Git
```

Comment:
I'd actually remove this one too, and possibly even config; should be passed to the functions if required as a param instead. Makes testing way easier as well.